### ISSUE TITLE
Better support for dataset labels in the config file

### ISSUE THREAD

**Issue by agitter** [Reed-CompBio/spras#16]
There is discussion about dataset labels in #12 that was not implemented.  This includes:
- make the dataset label field optional in the config file
- checking for duplicate dataset labels and automatically making the labels unique (with a warning)
- automatically assigning dataset labels for unlabeled datasets and warning the user
- logging dataset properties
- making the config file parsing robust to missing required fields

**Comment by annaritz** [comment]
There seems to be required filenames for dataset filenames. For example, there needs to be a "sources" and "targets" header. In `Datasets.py`, it appears that header names are stripped from the filename prefix in `load_files_from_dict()`.  We might want to standardize filenames too.

**Comment by agitter** [comment]
For the node properties, the primary way to specify attributes is to include a header row.  The column header becomes the attribute name.  This does rely on knowing which attributes each algorithm expects, which is not the most reliable approach.

A secondary way to specify attributes is a shorthand for Boolean attributes.  There is no header row, only a list of node ids that should all have the attribute set to True.  Because there is no header, the filename (without the extension suffix) is used as the attribute name.

We though the secondary format might be convenient for listing sources, targets, and similar node attributes.  Is it too confusing to have both formats?

This does need to be documented.  Should it go in the main readme for now?

**Comment by tristan-f-r** [comment]
I disagree with allowing optional dataset labels / making dataset labels unique. This requires that someone is actively monitoring the snakemake log for these warnings, which isn't something that one usually expects out of their users (especially with the massive amount of logs we produce).
